src/components/ShopCard/index.tsx

```diff
@@ -54,10 +54,22 @@ export default function ShopCard({ shop, ...rest }: Props) {
           </ListGroup.Item>
         </ListGroup>
         <Card.Body>
-          <Button variant="primary" block>
-            <FontAwesomeIcon icon={faCalendarCheck} fixedWidth />{' '}
-            {t('common:request-turn')}
-          </Button>
+          {shop.shopId ? (
+            <Button
+              variant="primary"
+              block
+              href={`/${shop.shopId}`}
+              disabled={!shop.isOpen}
+            >
+              <FontAwesomeIcon icon={faCalendarCheck} fixedWidth />{' '}
+              {t('common:request-turn')}
```

Comment:
I would also change the label to "closed now" or just "closed". That will give better feedback to the client why he is not available to request a turn.

Reply:
yes, thought about this one, let's change it for now and see what @grijos thinks